hathor/event/event_manager.py

```diff
@@ -42,9 +42,63 @@
     HathorEvents.REORG_FINISHED,
     HathorEvents.TX_METADATA_CHANGED,
     HathorEvents.BLOCK_METADATA_CHANGED,
+    HathorEvents.CONSENSUS_TX_UPDATE,
+    HathorEvents.CONSENSUS_TX_REMOVED,
 ]
 
 
+def _todo(data: Dict[str, Any]) -> Dict[str, Any]:
+    raise NotImplementedError('TODO')
+
+
+def _empty(data: Dict[str, Any]) -> Dict[str, Any]:
+    return {}
+
+
+def _extract_tx(data: Dict[str, Any]) -> Dict[str, Any]:
+    tx = data['tx']
+    return {
+        'hash': tx.hash_hex,
+        # TODO: the rest
+    }
+
+
+def _extract_reorg(data: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        'reorg_size': data['new_best_height'] - data['old_best_height'],
+        'previous_best_block': data['old_best_block'].hash_hex,
+        'new_best_block': data['new_best_block'].hash_hex,
+    }
+
+
+_EVENT_EXTRACT_MAP: Dict[HathorEvents, Callable[[Dict[str, Any]], Dict[str, Any]]] = {
+    HathorEvents.LOAD_STARTED: _empty,
+    HathorEvents.LOAD_FINISHED: _empty,
+    HathorEvents.NETWORK_NEW_TX_ACCEPTED: _extract_tx,
+    HathorEvents.NETWORK_NEW_TX_VOIDED: _extract_tx,
+    HathorEvents.NETWORK_BEST_BLOCK_FOUND: _extract_tx,
+    HathorEvents.NETWORK_ORPHAN_BLOCK_FOUND: _extract_tx,
+    HathorEvents.REORG_STARTED: _empty,
+    HathorEvents.REORG_FINISHED: _extract_reorg,
+    HathorEvents.TX_METADATA_CHANGED: _todo,  # XXX: I'm considering removing this event
+    HathorEvents.BLOCK_METADATA_CHANGED: _todo,  # XXX: I'm considering removing this event
+    HathorEvents.CONSENSUS_TX_UPDATE: _extract_tx,
+    HathorEvents.CONSENSUS_TX_REMOVED: _extract_tx,
+}
+
+
+def build_event_data(event: HathorEvents, event_args: EventArguments) -> Dict[str, Any]:
+    """Extract and build event data from event_args for a given event type."""
+    event_extract_fn = _EVENT_EXTRACT_MAP.get(event)
+    if event_extract_fn is None:
+        raise ValueError(f'The given event type ({event}) is not a supported event')
+    try:
+        return event_extract_fn(event_args.__dict__)
+    except NotImplementedError:
+        print(event)
```

Comment:
Do you intend to keep this print? If so, shouldn't we use a logger?

Reply:
I was using this for debugging. I'll remove it, I think any errors should bubble up and not be handled/logged around here.